### ISSUE TITLE
Crash: - AWSPinpointSessionClient endCurrentSessionTimeoutWithTimer

### ISSUE THREAD

**Issue by steveRobert** [aws/aws-sdk-ios#826]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?
Pinpoint

* In what version of SDK are you facing the problem?
2.6.10

* Is the issue limited to Simulators / Actual Devices?
Can't reproduce locally in the simulator or on device, but reported by Crashlytics (production - App Store)

* Can your problem be resolved if you bump to a higher version of SDK?
No

* Is this problem related to specific iOS version?
No (iOS 10 & 11)

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
Cocoa pods

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.
-

Here is my screenshot on fabric.io:
![awspinpoint-crash-reporting-crashlytics](https://user-images.githubusercontent.com/2506948/35912836-06163696-0bfe-11e8-9e12-c1126ee76361.png)

**Comment by steveRobert** [comment]
Hi @kvasukib ,
I use Pinpoint in basic configuration for tracking events and send push notifications for my users.
I have 2000+ users in my application, and last week, crashlytics notified me for a pinpoint crash for one user. I use 2.6.10 AWS SDK. 

Here is my simple code for AWS:
```
import Foundation
import AWSCore
import AWSPinpoint

struct TabacAWS {
    
    // AWS Debug
    private static var _debug: Bool = true
    static var debug: Bool {
        get { return _debug }
        set {
            _debug = newValue
            AWSDDLog.sharedInstance.logLevel = _debug ? AWSDDLogLevel.info : AWSDDLogLevel.off
            AWSDDLog.add(AWSDDTTYLogger.sharedInstance)
        }
    }
    
    // CredentialProvider
    static var credentialProvider: AWSCognitoCredentialsProvider? {
        return AWSCognitoCredentialsProvider(regionType: Constantes.AWS_CREDENTIAL_REGION, identityPoolId: Constantes.AWS_IDENTITY_POOL_ID)
    }
    
    // ServiveConfiguration
    static var serviceConfiguration: AWSServiceConfiguration? {
        return AWSServiceConfiguration(region: Constantes.AWS_SERVICE_REGION, credentialsProvider: credentialProvider)
    }
    
    // Pinpoint
    static private(set) var pinpoint: AWSPinpoint?
    @discardableResult static func pinpoint(with launchOptions: [AnyHashable :Any]?) ->AWSPinpoint? {
        let configuration = AWSPinpointConfiguration(appId: Constantes.AWS_APP_ID, launchOptions: launchOptions)
        configuration.serviceConfiguration = serviceConfiguration
        configuration.targetingServiceConfiguration = serviceConfiguration
        let value = AWSPinpoint(configuration: configuration)
        pinpoint = value
        return value
    }
   
    // Register Device Token
    static func registerDevice(token: Data) {
        pinpoint?.notificationManager.interceptDidRegisterForRemoteNotifications(withDeviceToken: token)
    }
    
    // HandleRemoteNotification
    static func handleRemoteNotification(_ userInfo: [AnyHashable: Any]) {
        pinpoint?.notificationManager.interceptDidReceiveRemoteNotification(userInfo, fetchCompletionHandler: { _ in })
    }
    
    // Event
    static func log(event: String, name: String, key: String) {
        if let client = pinpoint?.analyticsClient {
            let event = client.createEvent(withEventType: event)
            event.addAttribute(name, forKey: key)
            client.record(event)
            client.submitEvents()
        }
    }
}
```

**Comment by petrpavlik** [comment]
@kvasukib hi, has this been fixed yet? I couldn't find any release notes.

**Comment by petrpavlik** [comment]
@kvasukib Any news on this one? We're getting a significant amount of crashes with SDK 2.6.13 (the most recent one at this point) when the app goes to the background.

**Comment by itsmeichigo** [comment]
I actually had to set `enableAutoSessionRecording` to `NO` and handle the sessions myself to get rid of all the crashes.

**Comment by petrpavlik** [comment]
Thx @itsmeichigo! I'll have a look at that.

**Comment by kvasukib** [comment]
Hi @petrpavlik @itsmeichigo @steveRobert This has been fixed with 2.6.14 (https://github.com/aws/aws-sdk-ios/blob/master/CHANGELOG.md#2614). Please upgrade, check and let us know if it fixes the issue.

**Comment by petrpavlik** [comment]
Hi @kvasukib we finally have a new build with 2.6.16 in production and it looks this is now fixed. Thanks!
